NLog.Web.ASPNET5/LayoutRenderers/AspNetRequestUrlRenderer.cs

```diff
@@ -0,0 +1,91 @@
+
+namespace NLog.Web.LayoutRenderers
+{
+    /// <summary>
+    /// ASP.NET Request URL
+    /// </summary>
+    /// <para>Example usage of ${aspnet-request-url}:</para>
+    /// <example>
+    /// <code lang="NLog Layout Renderer">
+    /// ${aspnet-request-url:IncludeQueryString=true} - produces http://www.exmaple.com/?t=1
+    /// ${aspnet-request-url:IncludeQueryString=false} - produces http://www.exmaple.com/
+    /// ${aspnet-request-url:IncludePort=true} - produces http://www.exmaple.com:80/
+    /// ${aspnet-request-url:IncludePort=false} - produces http://www.exmaple.com/
+    /// ${aspnet-request-url:IncludePort=true:IncludeQueryString=true} - produces http://www.exmaple.com:80/?t=1
+    /// IncludePort - Is supported only NON ASP.NET CORE Version. After ASP.NET CORE RC2 Release support will be provided for ASP.NET Core as well.
+    /// </code>
+    /// </example>
+    [LayoutRenderer("aspnet-request-url")]
+    public class AspNetRequestUrlRenderer : AspNetLayoutRendererBase
+    {
+        /// <summary>
+        /// To specify whether to exclude / include the Query string.
+        /// </summary>
+        public bool IncludeQueryString { get; set; } = false;
+
+#if !DNX
+        /// <summary>
+        /// To specify whether to exclude / include the Port.
+        /// </summary>
+        public bool IncludePort { get; set; } = false;
+#endif
+        /// <summary>
+        /// Renders the Request URL from the HttpRequest
+        /// </summary>
+        /// <param name="builder">The <see cref="StringBuilder"/> to append the rendered data to.</param>
+        /// <param name="logEvent">Logging event.</param>
+        protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
+        {
+            var httpRequest = HttpContextAccessor.HttpContext.Request;
+
+            if (httpRequest == null)
+            {
+                return;
+            }
+            string url = String.Empty;
+            string pathAndQuery = String.Empty;
+            string port = String.Empty;
+#if !DNX
+            if (httpRequest.Url != null)
+            {
+                if (IncludePort && httpRequest.Url?.Port > 0)
+                {
+                    port = ":" + httpRequest.Url.Port.ToString();
+                }
+
+                if (IncludeQueryString && httpRequest.Url?.PathAndQuery != null)
+                {
+                    pathAndQuery = httpRequest.Url.PathAndQuery;
+                }
+                else
+                {
+                    pathAndQuery = httpRequest.Url.AbsolutePath;
+                }
+
+                url = $"{httpRequest.Url.Scheme}://{httpRequest.Url.Host}{port}{pathAndQuery}";
+            }
+
+#else
+            if (IncludeQueryString)
+                pathAndQuery = httpRequest.QueryString.ToUriComponent();
+
+            url = $"{httpRequest.Scheme}://{httpRequest.Host.ToUriComponent()}{httpRequest.PathBase.ToUriComponent()}{httpRequest.Path.ToUriComponent()}{pathAndQuery}";
```

Comment:
port is not supported in DNX?

I think it's maybe more clear to create more variables, and move the `url=$""` outside the `#if` etc. What do you think?